remote_attestation/rust/src/tests/message.rs

```diff
@@ -0,0 +1,222 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{
+    crypto::{
+        KEY_AGREEMENT_ALGORITHM_KEY_LENGTH, NONCE_LENGTH, SIGNATURE_LENGTH,
+        SIGNING_ALGORITHM_KEY_LENGTH,
+    },
+    message::{
+        deserialize_message, ClientHello, ClientIdentity, Deserializable, EncryptedData,
+        MessageWrapper, Serializable, ServerIdentity, MAXIMUM_MESSAGE_SIZE,
+        REPLAY_PROTECTION_ARRAY_LENGTH,
+    },
+};
+use anyhow::{anyhow, Context};
+use assert_matches::assert_matches;
+use quickcheck::{quickcheck, TestResult};
+
+const INVALID_MESSAGE_HEADER: u8 = 5;
+
+/// Creates a zero initialized array.
+fn default_array<T, const L: usize>() -> [T; L]
+where
+    T: std::marker::Copy + std::default::Default,
+{
+    [Default::default(); L]
+}
+
+/// Converts slices to arrays (expands with zeroes).
+fn to_array<T, const L: usize>(input: &[T]) -> anyhow::Result<[T; L]>
+where
+    T: std::marker::Copy + std::default::Default,
+{
+    if input.len() <= L {
+        let mut result: [T; L] = default_array();
```

Comment:
Why is it not sufficient to do `let mut result: [T; L] = Default::default()` here? Is it not implemented for arrays?

Reply:
It is only implemented for specific sizes, i.e. 1,2,...,32

Reply:
That seems worth documenting, right?

Reply:
Done